packages/collector/test/tracing/cloud/aws/s3/test.js

```diff
@@ -0,0 +1,210 @@
+];
+
+if (!supportedVersion(process.versions.node)) {
+  mochaSuiteFn = describe.skip;
+} else {
+  mochaSuiteFn = describe;
+}
+
+const retryTime = config.getTestTimeout() * 2;
+
+mochaSuiteFn('tracing/cloud/aws/s3', function() {
+  this.timeout(config.getTestTimeout() * 3);
+
+  globalAgent.setUpCleanUpHooks();
+  const agentControls = globalAgent.instance;
+
+  describe('tracing enabled, no suppression', function() {
+    const appControls = new ProcessControls({
+      appPath: path.join(__dirname, 'app'),
+      port: 3215,
+      useGlobalAgent: true,
+      env: {
+        AWS_S3_BUCKET_NAME: bucketName
+      }
+    });
+
+    ProcessControls.setUpHooksWithRetryTime(retryTime, appControls);
+    withErrorOptions.forEach(withError => {
+      requestMethods.forEach(requestMethod => {
+        describe(`getting result with: ${requestMethod}; with error: ${withError ? 'yes' : 'no'}`, () => {
+          availableOperations.forEach(operation => {
+            it(`operation: ${operation}`, async () => {
+              const withErrorOption = withError ? '?withError=1' : '';
+              const apiPath = `/${operation}/${requestMethod}`;
+
+              const response = await appControls.sendRequest({
+                method: 'GET',
+                path: `${apiPath}${withErrorOption}`,
+                simple: withError === false
+              });
+
+              return verify(appControls, response, apiPath, withError);
+            });
+          });
+        });
+      });
+    });
+
+    function verify(controls, response, apiPath, withError) {
+      return retry(() => {
+        return agentControls.getSpans().then(spans => verifySpans(controls, spans, apiPath, withError));
+      }, retryTime);
+    }
+
+    function verifySpans(controls, spans, apiPath, withError) {
+      const httpEntry = verifyHttpEntry(spans, apiPath, controls);
+      verifyS3Exit(spans, httpEntry, withError);
+      verifyHttpExit(spans, httpEntry, controls, withError);
+    }
+
+    function verifyHttpEntry(spans, apiPath, controls) {
+      return expectExactlyOneMatching(spans, [
+        span => expect(span.p).to.not.exist,
+        span => expect(span.k).to.equal(constants.ENTRY),
+        span => expect(span.f.e).to.equal(String(controls.getPid())),
+        span => expect(span.f.h).to.equal('agent-stub-uuid'),
+        span => expect(span.n).to.equal('node.http.server'),
+        span => expect(span.data.http.url).to.equal(apiPath)
+      ]);
+    }
+
+    function verifyHttpExit(spans, parentSpan, controls, withError) {
+      return expectExactlyOneMatching(spans, [
+        span => expect(span.t).to.equal(parentSpan.t),
+        span => expect(span.p).to.equal(parentSpan.s),
+        span => expect(span.k).to.equal(constants.EXIT),
+        span => expect(span.f.e).to.equal(String(controls.getPid())),
+        span => expect(span.f.h).to.equal('agent-stub-uuid'),
+        span => expect(span.n).to.equal(withError ? 's3' : 'node.http.client')
```

Comment:
This is weird and a bit confusing. Why is this checking for an s3 span in the error test cases?